pubtools/pulplib/_impl/client/search.py

```diff
@@ -21,6 +23,15 @@ def all_subclasses(klass):
     return out
 
 
+def to_mongo_json(value):
+    # Return a value converted to the format expected for a mongo JSON
+    # expression. Only a handful of special types need explicit conversions.
+    if isinstance(value, datetime.datetime):
+        return {"$date": value.strftime("%Y-%m-%dT%H:%M:%SZ")}
+
```

Comment:
Looks like this was oversimplified and should also recurse into lists and dicts, something like this:

```suggestion
    if isinstance(value, (list, tuple)):
        return [to_mongo_json(elem) for elem in value]
    if isinstance(value, dict):
        out = {}
        for (key, child_value) in value.items():
            out[key] = to_mongo_json(child_value)
        return out
```

So that e.g. testing if a value is $in one of a series of dates will work.